src/shared/conf-parser.c

```diff
@@ -648,6 +649,35 @@ int config_parse_bool(const char* unit,
         return 0;
 }
 
+int config_parse_id128(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        sd_id128_t t, *result = data;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        r = sd_id128_from_string(rvalue, &t);
+        if (r < 0)
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse 128bit ID/UUID, ignoring: %s", rvalue);
+        else if (sd_id128_is_null(t))
+                log_syntax(unit, LOG_ERR, filename, line, r, "128bit ID/UUID is all 0, ignoring: %s", rvalue);
```

Comment:
hmm, i guess this call should pass the 5th arg as 0 instead of r, after all r is not really set to an error here (unlike in the first if branch), and you are supposed to pass the error that occured there.  That said, r happens to be zero, so actually passing r and 0 is actually equivalent.

Hence, I guess the current code is OK, but with an explicit 0 I think it would be a lot clearer.

Reply:
I am really impressed with how thorough your code review is! Thank you for taking that much time to read through my C scribblings!

I hope that did not sound sarcastic, it is absolutely not intended that way. I am well aware how much time and effort code reviews are.

Reply:
thanks! we are trying...

Thank you for your PR! Much appreciated!